File: java-patterns/src/main/java/behavioral/mediator/Colleague.java
package behavioral.mediator;

/**
 * Usage: 同事接口 <br/>
 * Date: 2023/5/13 16:42 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public interface Colleague {
    void receive();

    void send();

    void setMedium(Mediator mediator);
}

File: go-patterns/behavioral/mediator/colleague.go
package mediator

import "fmt"

type Colleague interface {
	receive()
	send()
	setMedium(m Mediator)
}

type FooColleague struct {
	Mediator
}

func (fc *FooColleague) receive() {
	fmt.Println("FooColleague.receive")
}

func (fc *FooColleague) send() {
	fmt.Println("FooColleague.send")
	fc.Mediator.relay(fc)
}

func (fc *FooColleague) setMedium(m Mediator) {
	fc.Mediator = m
}

type BarColleague struct {
	Mediator
}

func (bc *BarColleague) receive() {
	fmt.Println("BarColleague.receive")
}

func (bc *BarColleague) send() {
	fmt.Println("BarColleague.send")
	bc.relay(bc)
}

func (bc *BarColleague) setMedium(m Mediator) {
	bc.Mediator = m
}
